### DISCUSSION TITLE
youtube / nativereplace scriptlet

### DISCUSSION THREAD

**Discussion by ryanbr** [uBlockOrigin/uBlock-issues#3269] [Ideas]
Similar functionality to replace= but works on other browsers. Just brainstorming to improve the current issue with youtube SSAP Ad situation.

**Comment by stephenhawk8054** [comment]
Modifying the responses before the browser parsing the them needs to be supported by the browser itself: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/filterResponseData

`rpnt` is actually an attempt for that, but at the end scriptlets are injected together with the websites loaded, its core problem is it's impossible to have a reliable one.

**Reply by ryanbr** [reply]
Maybe more of a hack solution is needed here, to get around the browser limitation. We just need something, anything to improve this youtube issue 

> i'm saying we need something upstream, even its more hacks (changing rpnt, or new scriptlet). since YT have basically found a flaw here. Fixing it for FF for the 5% still using FF is an imperfect solution tbh.

**Reply by gorhill** [reply]
`rpnt` is the hack solution. The `scripting` API could help making the scriptlet more reliable but it's only available in MV3 for chromium-based browsers.
